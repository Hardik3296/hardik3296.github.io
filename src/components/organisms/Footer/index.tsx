import { Dispatch, SetStateAction, useContext } from "react";
import { ThemeContext, ThemeInterface } from "../../../utils/contexts/ThemeContext";
import styles from "./styles.module.scss";
import Icons8SVG from "../../../assets/svgs/Icons8SVG";
import MaterialUISVG from "../../../assets/svgs/MaterialUISVG";
import ReactSVG from "../../../assets/svgs/ReactSVG";
import SassSVG from "../../../assets/svgs/SassSVG";


const Footer = (): JSX.Element => {

   const [theme, _] = useContext<[ThemeInterface, Dispatch<SetStateAction<ThemeInterface>>]>(ThemeContext);

   const handleCLick = (link: string): void => {
      window.open(link, "_blank");
   }

   return (
      <div className={styles.container} style={{ backgroundColor: theme.repos.div.background }}>
         <p style={theme.text}>Made with </p>
         <ReactSVG height={50} width={50} className={styles.svg} onClick={() => handleCLick("https://reactjs.org/")} />
         <SassSVG height={50} width={50} className={styles.svg} onClick={() => handleCLick("https://sass-lang.com/")} />
         <MaterialUISVG height={50} width={50} className={styles.svg} onClick={() => handleCLick("https://material-ui.com/")} />
         <Icons8SVG height={50} width={50} className={styles.svg} onClick={() => handleCLick("https://icons8.com/")} />
      </div>
   );
}

export default Footer;