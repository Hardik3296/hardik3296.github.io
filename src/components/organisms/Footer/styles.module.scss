@import "../../../fonts.scss";

.container{
   display:flex;
   flex-direction:row;
   align-items: center;
   justify-content: center;
   padding:1vh 15px;
   p{
      font-family: ArchivoMedium;
      margin-right:20px;
   }
   .svg{
      margin-right:2vw;
      will-change: transform;
      &:hover{
         @media(min-width:1025px){
            cursor: pointer;
            transform: scale(1.2);
            transition: transform 0.5s ease-in-out;
         }
        
      }
   }
}