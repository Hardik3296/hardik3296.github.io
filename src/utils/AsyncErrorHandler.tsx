import React from "react";

const useAsyncError = () => {
   const [_error, setError] = React.useState();
   return React.useCallback(
      e => {
         setError(() => {
            throw e;
         });
      },
      [setError],
   );
};

export default useAsyncError;