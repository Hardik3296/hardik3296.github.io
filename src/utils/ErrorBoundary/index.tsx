import { Component, ReactNode } from "react";
import styles from "./styles.module.scss";

interface MyProps {
   textcolor: string,
   children: ReactNode,
}

class ErrorBoundary extends Component<MyProps, {}> {

   state = {
      hasError: false
   }

   static getDerivedStateFromError() {
      return { hasError: true }
   }

   componentDidCatch(error: any, errorInfo: any) {
   }
   render() {
      if (this.state.hasError)
         return <div className={styles.errorDiv}>
            <p style={{ color: this.props.textcolor }}>Something went wrong while fetching the repositories!!</p>
         </div>
      else
         return this.props.children;
   }
}

export default ErrorBoundary;