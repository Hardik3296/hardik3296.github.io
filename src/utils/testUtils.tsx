import { ShallowWrapper } from "enzyme"

export const findAttributeByTestId = (component:ShallowWrapper, attr:string):ShallowWrapper=>{
   return component.find(`[test-id='${attr}']`);
}