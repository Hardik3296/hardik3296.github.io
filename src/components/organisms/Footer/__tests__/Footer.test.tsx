import Footer from "..";
import { shallow, ShallowWrapper } from "enzyme";
import { findAttributeByTestId } from "../../../../utils/testUtils";
import renderer from "react-test-renderer";

const renderComponent = (): ShallowWrapper => {
   return shallow(<Footer />)
}

describe("Footer component", () => {
   let wrapper: ShallowWrapper;

   beforeEach(() => {
      wrapper = renderComponent();
   });

   it("Should contain 1 react icon", () => {
      const component = findAttributeByTestId(wrapper, "react-icon");
      expect(component.length).toBe(1);
   });

   it("Should contain 1 typescript icon", () => {
      const component = findAttributeByTestId(wrapper, "typescript-icon");
      expect(component.length).toBe(1);
   });

   it("Should contain 1 material icon", () => {
      const component = findAttributeByTestId(wrapper, "material-icon");
      expect(component.length).toBe(1);
   });

   it("Should contain 1 icons8 icon", () => {
      const component = findAttributeByTestId(wrapper, "icons8-icon");
      expect(component.length).toBe(1);
   });

   it("Should contain 1 sass icon", () => {
      const component = findAttributeByTestId(wrapper, "sass-icon");
      expect(component.length).toBe(1);
   });

   it("Should render correctly", () => {
      const component = renderer.create(<Footer />).toJSON();
      expect(component).toMatchSnapshot();
   });
});