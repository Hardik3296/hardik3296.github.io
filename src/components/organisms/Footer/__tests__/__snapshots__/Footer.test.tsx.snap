// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Should render correctly 1`] = `
<div
  className="container"
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <p
    style={
      Object {
        "color": "#000000",
      }
    }
    test-id="text"
  >
    Made with 
  </p>
  <svg
    className="svg"
    height={50}
    onClick={[Function]}
    test-id="react-icon"
    viewBox="0 0 100 100"
    width={50}
  >
    <path
      d="M69.11 7.969c-1.56-.051-3.192.218-4.829.75-3.273 1.062-6.703 3.129-10.222 5.992-1.38 1.121-2.77 2.406-4.168 3.766-1.368-1.329-2.73-2.575-4.078-3.672-3.516-2.86-6.938-4.93-10.204-5.989-3.265-1.058-6.515-1.09-9.183.45-2.668 1.539-4.266 4.367-4.98 7.726-.716 3.36-.637 7.356.081 11.832.25 1.555.606 3.188 1.012 4.848a60.42 60.42 0 0 0-4.379 1.484c-4.144 1.617-7.574 3.555-10.07 5.86-2.496 2.304-4.11 5.101-4.11 8.172 0 3.07 1.614 5.863 4.11 8.167 2.496 2.305 5.926 4.247 10.07 5.864 1.469.574 3.059 1.09 4.7 1.578-.59 2.207-1.07 4.348-1.395 6.375-.719 4.48-.797 8.48-.078 11.844.718 3.363 2.312 6.21 4.996 7.757 2.683 1.547 5.945 1.512 9.219.45 3.273-1.063 6.699-3.125 10.218-5.989 1.367-1.113 2.75-2.379 4.133-3.726 1.39 1.355 2.777 2.629 4.152 3.746 3.516 2.86 6.942 4.926 10.208 5.984 3.265 1.059 6.515 1.09 9.183-.449 2.668-1.539 4.262-4.367 4.977-7.727.715-3.359.636-7.351-.082-11.828-.329-2.035-.805-4.183-1.399-6.394 1.695-.5 3.336-1.031 4.848-1.621 4.144-1.617 7.574-3.559 10.07-5.864 2.496-2.304 4.11-5.097 4.11-8.167 0-3.07-1.614-5.868-4.11-8.172-2.496-2.305-5.926-4.243-10.07-5.86-1.395-.543-2.895-1.035-4.445-1.504.41-1.672.773-3.316 1.023-4.879.719-4.48.797-8.484.078-11.847-.719-3.363-2.316-6.207-5-7.754-1.34-.774-2.828-1.152-4.387-1.203zm-.184 4.07c.98.027 1.812.258 2.527.672 1.43.824 2.484 2.43 3.047 5.07.563 2.64.543 6.211-.121 10.348-.227 1.418-.559 2.918-.934 4.457-4.09-.984-8.574-1.715-13.351-2.145-2.399-3.324-4.864-6.351-7.348-9.035 1.313-1.281 2.621-2.484 3.895-3.52 3.25-2.644 6.328-4.445 8.898-5.277 1.285-.418 2.406-.597 3.387-.57zm-37.953.012c.984-.028 2.109.152 3.398.57 2.578.836 5.66 2.633 8.914 5.281 1.25 1.016 2.528 2.192 3.813 3.446-2.504 2.707-4.989 5.761-7.403 9.113-4.758.437-9.222 1.172-13.293 2.16-.37-1.531-.703-3.023-.93-4.434-.663-4.14-.683-7.71-.117-10.359.567-2.648 1.63-4.265 3.07-5.098.72-.418 1.563-.652 2.548-.68zm18.98 12.234a91.115 91.115 0 0 1 4.875 5.8A115.47 115.47 0 0 0 50 29.98c-1.664 0-3.305.04-4.922.11a90.141 90.141 0 0 1 4.875-5.805zM50 34.02c2.719 0 5.371.113 7.95.3a113.33 113.33 0 0 1 5.16 8.114 113.444 113.444 0 0 1 3.96 7.504 113.822 113.822 0 0 1-4.015 7.605 115.753 115.753 0 0 1-4.024 6.43c-2.914.242-5.93.379-9.031.379-3.176 0-6.262-.145-9.238-.399a113.647 113.647 0 0 1-3.957-6.332 114.144 114.144 0 0 1-4.004-7.59c1.21-2.52 2.547-5.074 4.023-7.633a113.024 113.024 0 0 1 5.137-8.07c2.605-.191 5.289-.308 8.039-.308zm13.098.8c3.297.41 6.39.985 9.273 1.668a89.834 89.834 0 0 1-3.183 8.75 121.53 121.53 0 0 0-2.614-4.804 121.368 121.368 0 0 0-3.476-5.614zm-26.41.032a122.083 122.083 0 0 0-3.399 5.503 119.874 119.874 0 0 0-2.652 4.868 89.736 89.736 0 0 1-3.16-8.7 86.127 86.127 0 0 1 9.21-1.671zm39.64 2.699c1.418.43 2.785.875 4.043 1.367 3.82 1.492 6.844 3.258 8.8 5.066 1.958 1.809 2.81 3.532 2.81 5.203 0 1.672-.852 3.395-2.81 5.204-1.956 1.808-4.98 3.574-8.8 5.066-1.394.547-2.93 1.04-4.523 1.508-1.137-3.516-2.59-7.188-4.317-10.934 1.985-4.3 3.59-8.496 4.797-12.48zm-52.723.02c1.192 3.945 2.778 8.097 4.735 12.355-1.738 3.77-3.2 7.457-4.34 10.996-1.54-.457-3.02-.938-4.371-1.465-3.82-1.492-6.844-3.258-8.8-5.066-1.958-1.809-2.81-3.532-2.81-5.203 0-1.672.852-3.395 2.81-5.204 1.956-1.808 4.98-3.574 8.8-5.066 1.238-.484 2.582-.926 3.976-1.348zM50 40c-5.5 0-10 4.5-10 10s4.5 10 10 10 10-4.5 10-10-4.5-10-10-10zM30.684 54.73c.84 1.625 1.714 3.258 2.66 4.891a123.476 123.476 0 0 0 2.258 3.746 84.92 84.92 0 0 1-7.637-1.402 93.113 93.113 0 0 1 2.719-7.235zm38.554 0a93.556 93.556 0 0 1 2.727 7.25 85.503 85.503 0 0 1-7.66 1.399 114.722 114.722 0 0 0 2.289-3.793 119.25 119.25 0 0 0 2.644-4.856zM26.813 65.848c3.62.859 7.542 1.515 11.703 1.945 2.777 4.016 5.672 7.625 8.582 10.781-1.301 1.27-2.594 2.461-3.856 3.489-3.25 2.644-6.332 4.44-8.902 5.273-2.57.832-4.485.723-5.914-.102-1.43-.824-2.48-2.43-3.043-5.07-.563-2.64-.547-6.207.117-10.344.3-1.879.754-3.886 1.313-5.972zm46.308.015c.567 2.098 1.016 4.117 1.32 6.004.664 4.14.684 7.711.118 10.36-.567 2.648-1.621 4.27-3.063 5.101-1.441.832-3.375.942-5.953.106-2.578-.836-5.66-2.633-8.914-5.282-1.274-1.035-2.57-2.234-3.883-3.515 2.93-3.168 5.844-6.797 8.64-10.836 4.173-.426 8.106-1.082 11.735-1.938zm-29.387 2.34c2.055.113 4.133.188 6.266.188 2.063 0 4.074-.07 6.063-.176a92.227 92.227 0 0 1-6.168 7.48 92.442 92.442 0 0 1-6.16-7.492zm0 0"
      style={
        Object {
          "fill": "#6c63ff",
          "fillOpacity": 1,
          "fillRule": "evenodd",
          "stroke": "none",
        }
      }
    />
  </svg>
  <svg
    className="svg"
    height={50}
    onClick={[Function]}
    test-id="sass-icon"
    viewBox="0 0 96 96"
    width={50}
  >
    <path
      d="M79.734 51.91c-3.078.02-5.742.754-7.972 1.86-.817-1.633-1.645-3.067-1.782-4.133-.164-1.242-.351-1.989-.156-3.469.2-1.48 1.055-3.582 1.043-3.738-.012-.16-.191-.914-1.965-.926-1.773-.016-3.293.34-3.468.809-.18.468-.524 1.523-.735 2.62-.308 1.61-3.543 7.348-5.375 10.348-.601-1.172-1.11-2.207-1.215-3.02-.164-1.245-.351-1.988-.156-3.472.195-1.48 1.055-3.582 1.043-3.738-.012-.16-.191-.914-1.965-.926-1.773-.012-3.293.34-3.472.809-.176.468-.368 1.562-.73 2.62-.364 1.06-4.657 10.63-5.786 13.11a75.657 75.657 0 0 1-1.422 2.973c-.004 0-.004-.004-.004-.004s-.02.047-.058.125c-.301.59-.48.918-.48.918l.007.011c-.238.434-.496.836-.621.836-.09 0-.266-1.156.039-2.738.637-3.32 2.172-8.496 2.156-8.676-.008-.093.29-.996-1-1.468-1.254-.457-1.7.308-1.812.308-.11.004-.192.27-.192.27s1.395-5.82-2.66-5.82c-2.535 0-6.047 2.773-7.777 5.288-1.094.594-3.43 1.872-5.91 3.235-.95.523-1.922 1.055-2.844 1.562a9.237 9.237 0 0 0-.192-.207c-4.918-5.246-14.003-8.957-13.62-16.007.14-2.567 1.03-9.317 17.464-17.504 13.461-6.711 24.242-4.864 26.102-.774C56.879 28.84 48.46 39.7 34.48 41.27c-5.328.597-8.132-1.47-8.832-2.239-.734-.808-.843-.843-1.117-.691-.445.246-.164.96 0 1.387.418 1.086 2.13 3.011 5.051 3.972 2.57.844 8.824 1.305 16.387-1.62 8.472-3.278 15.086-12.392 13.14-20.013-1.976-7.746-14.824-10.293-26.984-5.972-7.246 2.57-15.078 6.605-20.715 11.875-6.695 6.261-7.765 11.718-7.324 13.996 1.562 8.094 12.723 13.367 17.191 17.273-.222.121-.43.235-.617.34-2.238 1.11-10.746 5.559-12.875 10.262-2.414 5.336.387 9.164 2.238 9.68 5.739 1.593 11.625-1.278 14.79-5.997 3.164-4.718 2.78-10.867 1.328-13.668-.02-.035-.04-.07-.055-.105.574-.344 1.168-.691 1.75-1.035a119.07 119.07 0 0 1 3.23-1.82c-.543 1.488-.941 3.273-1.148 5.851-.242 3.027.996 6.941 2.621 8.48.719.68 1.578.696 2.121.696 1.89 0 2.75-1.57 3.703-3.434 1.164-2.277 2.196-4.933 2.196-4.933s-1.293 7.172 2.238 7.172c1.289 0 2.582-1.672 3.16-2.52v.012l.102-.164c.132-.203.21-.336.21-.336s0-.012.004-.035c.52-.899 1.668-2.95 3.387-6.325 2.223-4.367 4.356-9.832 4.356-9.832s.199 1.336.847 3.547c.383 1.301 1.196 2.739 1.836 4.117-.515.715-.832 1.125-.832 1.125s.004.008.008.024c-.41.547-.871 1.137-1.355 1.715-1.758 2.09-3.848 4.476-4.13 5.164-.328.812-.25 1.41.387 1.89.465.352 1.293.407 2.16.348 1.579-.105 2.688-.496 3.235-.734a10.915 10.915 0 0 0 2.781-1.461c1.723-1.27 2.758-3.078 2.66-5.477-.058-1.324-.476-2.633-1.011-3.87.156-.224.312-.45.468-.68 2.715-3.965 4.82-8.329 4.82-8.329s.2 1.336.848 3.551c.328 1.117.98 2.34 1.563 3.535-2.551 2.074-4.133 4.485-4.684 6.063-1.015 2.925-.219 4.25 1.274 4.55.675.137 1.632-.171 2.347-.476.895-.297 1.969-.79 2.973-1.527 1.723-1.266 3.379-3.04 3.277-5.438-.043-1.094-.34-2.176-.742-3.215 2.164-.902 4.965-1.402 8.531-.988 7.657.894 9.157 5.672 8.868 7.672-.286 2-1.891 3.101-2.43 3.433-.535.332-.7.45-.656.696.066.355.316.343.773.265.629-.105 4.02-1.625 4.164-5.316.184-4.691-4.308-9.84-12.262-9.8zM20.738 71.8c-2.535 2.766-6.074 3.81-7.593 2.93-1.641-.949-.993-5.023 2.12-7.957 1.895-1.789 4.344-3.437 5.965-4.453.372-.218.915-.547 1.575-.941l.172-.098-.004-.004c.129-.074.257-.152.394-.234 1.133 4.176.035 7.855-2.629 10.758zm18.473-12.554c-.883 2.152-2.73 7.66-3.856 7.363-.968-.254-1.554-4.437-.195-8.558.688-2.074 2.149-4.555 3.012-5.516 1.383-1.547 2.906-2.055 3.277-1.426.469.801-1.683 6.786-2.238 8.137zm15.258 7.281c-.375.196-.719.317-.875.223-.121-.07.152-.328.152-.328s1.906-2.055 2.66-2.988c.438-.543.946-1.192 1.496-1.91.004.074.004.144.004.214-.004 2.461-2.379 4.121-3.437 4.79zm11.754-2.68c-.282-.199-.235-.84.683-2.843.364-.785 1.188-2.11 2.614-3.371.168.52.265 1.015.265 1.48-.02 3.098-2.226 4.254-3.562 4.735zm0 0"
      style={
        Object {
          "fill": "#6c63ff",
          "fillOpacity": 1,
          "fillRule": "nonzero",
          "stroke": "none",
        }
      }
    />
  </svg>
  <svg
    className="svg"
    height={50}
    onClick={[Function]}
    test-id="material-icon"
    viewBox="0 0 96 96"
    width={50}
  >
    <path
      d="m2 10 12 8v40L2 50zm92 30-12 8v26l12-8zm0-28-12 8v14l12-8zm-24-2-12 8v40l12-8zm0 0"
      style={
        Object {
          "fill": "#6c63ff",
          "fillOpacity": 1,
          "fillRule": "nonzero",
          "stroke": "none",
        }
      }
    />
    <path
      d="m2 24 34 22V32L2 10zm68 0L36 46V32l34-22zm0 28L36 74V60l34-22zm24 16L60 90V76l34-22zm0 0"
      style={
        Object {
          "fill": "#6c63ff",
          "fillOpacity": 1,
          "fillRule": "nonzero",
          "stroke": "none",
        }
      }
    />
    <path
      d="M60 75.531 36 60v14l24 15.531zm0 0"
      style={
        Object {
          "fill": "#6c63ff",
          "fillOpacity": 1,
          "fillRule": "nonzero",
          "stroke": "none",
        }
      }
    />
  </svg>
  <svg
    className="svg"
    height={50}
    onClick={[Function]}
    test-id="icons8-icon"
    viewBox="0 0 60 60"
    width={50}
  >
    <path
      d="M10.445 8A2.444 2.444 0 0 0 8 10.445v39.11A2.444 2.444 0 0 0 10.445 52h17.11A2.444 2.444 0 0 0 30 49.555v-39.11A2.444 2.444 0 0 0 27.555 8zM30 19c0 6.066 4.934 11 11 11s11-4.934 11-11S47.066 8 41 8s-11 4.934-11 11zm11 11c-6.066 0-11 4.934-11 11s4.934 11 11 11 11-4.934 11-11-4.934-11-11-11zm0 0"
      style={
        Object {
          "fill": "#6c63ff",
          "fillOpacity": 1,
          "fillRule": "nonzero",
          "stroke": "none",
        }
      }
    />
  </svg>
  <svg
    className="svg"
    height={50}
    onClick={[Function]}
    test-id="typescript-icon"
    viewBox="0 0 100 100"
    width={50}
  >
    <path
      d="M10 8a2 2 0 0 0-2 2v80a2 2 0 0 0 2 2h80a2 2 0 0 0 2-2V10a2 2 0 0 0-2-2zm2 4h76v76H12zm18 34v6.89h10V84h8V52.89h10V46zm43.383.02c-6.23-.024-13.348 1.863-13.348 10.628 0 11.461 15.442 11.461 15.442 16.633 0 .489.207 4.051-5.243 4.051-5.449 0-9.992-3.426-9.992-3.426v8.317s23.762 7.683 23.762-9.645C84 61.328 68.418 61.891 68.418 56.301c0-2.164 1.539-4.192 5.8-4.192 4.262 0 8.036 2.516 8.036 2.516l.281-7.41s-4.308-1.176-9.152-1.195zm0 0"
      style={
        Object {
          "fill": "#6c63ff",
          "fillOpacity": 1,
          "fillRule": "nonzero",
          "stroke": "none",
        }
      }
    />
  </svg>
</div>
`;
